app/src/main/java/com/keylesspalace/tusky/components/compose/ComposeActivity.kt

```diff
@@ -407,7 +407,11 @@ class ComposeActivity : BaseActivity(),
         // If you select "backward" in an editable, you get SelectionStart > SelectionEnd
         val start = composeEditField.selectionStart.coerceAtMost(composeEditField.selectionEnd)
         val end = composeEditField.selectionStart.coerceAtLeast(composeEditField.selectionEnd)
-        composeEditField.text.replace(start, end, text)
+        val textToInsert = if (
+                composeEditField.text.isNotEmpty()
+                && !composeEditField.text[start - 1].isWhitespace()
+        ) " $text" else text
+        composeEditField.text.replace(start, end, textToInsert)
```

Comment:
This is also used for the #/@ special character buttons - does this change also make sense for those?

Reply:
I've seen it and I think so? They are not pasted as special sequences without leading space, at least by our editor